ui/plugin-system/src/runtime/TimeRangeProvider/TimeRangeProvider.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 The Perses Authors
```

Comment:
Moved all time range related code from 'dashboard' to `plugin-system`

Reply:
Awesome that this worked without creating a circular dependency 🎉  This does mean that use-query-params become a required dependency for plugin-system, but I think that's worthwhile to not split the implementation b/w packages, can you update the plugin-system package.json?

Reply:
Yeah, I don't love that plugin authors now have to take that dependency through `plugin-system`, but I think we're gonna be figuring out how to split this stuff up differently pretty soon with the testing convo.